cli/src/commands/parse/pst.rs

```diff
@@ -0,0 +1,1275 @@
+use anyhow::{anyhow, Context, Result};
+use chrono::{DateTime, TimeZone, Utc};
+use colored::Colorize;
+use itertools::Itertools;
+use libc::{c_char, c_int, size_t};
+use mailparse::{parse_header, MailHeader, MailHeaderMap};
+use reinfer_client::{
+    resources::{
+        attachments::AttachmentMetadata,
+        email::{EmailMetadata, Id as EmailId},
+    },
+    BucketIdentifier, Client, Mailbox, MimeContent, NewEmail,
+};
+use std::{
+    collections::HashSet,
+    path::{Path, PathBuf},
+    sync::{
+        atomic::{AtomicUsize, Ordering},
+        Arc,
+    },
+};
+use structopt::StructOpt;
+
+use crate::progress::{Options as ProgressOptions, Progress};
+
+const ROOT_FOLDER_NAME: &str = "root";
+
+#[repr(C)]
+#[derive(Debug, Clone)]
+struct LibPffItem {
+    _unused: c_int,
+}
+
+#[repr(C)]
+#[derive(Debug, Clone)]
+struct LibPffRecordEntry {
+    _unused: c_int,
+}
+
+#[repr(C)]
+#[derive(Debug, Clone)]
+struct LibPffRecordSet {
+    _unused: c_int,
+}
+
+#[repr(C)]
+#[derive(Debug, Clone)]
+struct LibPffFile {
+    _unused: c_int,
+}
+
+#[repr(C)]
+#[derive(Debug, Clone)]
+struct LibcError {
+    number_of_messages: c_int,
+    messages: *const *const c_char,
+}
+
+#[derive(Clone)]
+enum LibPffEntryType {
+    MessageTransportHeaders = 0x007d,
+    AttachmentFilenameLong = 0x3707,
+}
+
+#[derive(Clone)]
+enum LibPffEntryValueFlags {
+    MatchAnyValueType = 0x01,
+}
+
+impl LibcError {
+    pub fn as_string(&self) -> String {
+        let mut errors: Vec<String> = Vec::new();
+        for i in 0..self.number_of_messages {
+            let message_ptr = unsafe { self.messages.offset(i as isize) };
+
+            if message_ptr.is_null() {
+                errors.push("Unknown Error: message_ptr null".to_string());
+                continue;
+            }
+
+            if !message_ptr.is_aligned() {
+                errors.push("Unknown Error: message_ptr misaligned".to_string());
+                continue;
+            }
+
+            let message = unsafe { std::ffi::CStr::from_ptr(*message_ptr) }
+                .to_string_lossy()
+                .into_owned();
```

Comment:
Generally for `unsafe` blocks, you should leave a comment explaining why it's safe (or assumptions you're making/invariants the compiler can't enforce). This is similar to `expect` or `unwrap` in safe code.

So here I'd say something like

```rust
// Assumes that the pointer returned from `offset` is valid
let message_ptr = unsafe { self.messages.offset(i as isize) };
...
// Assumes the pointer returned from libpff is a valid CString
let message = unsafe { std::ffi::CStr::from_ptr(*message_ptr) }
```
```